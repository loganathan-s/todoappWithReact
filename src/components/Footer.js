import React from 'react';

class Footer extends React.Component{
	render(){
		return(
			<div className="footer">
				<footer className="footer">
					<p> Footer</p>
				</footer>
			</div>
			)
	}

}

export default Footer;

