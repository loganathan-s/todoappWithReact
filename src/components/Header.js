import React from 'react';

class Header extends React.Component{
	render(){
		return(
			<div className="header">
				<header className="top">
					<h1> This is a simple TODO application, developed using React 15.6 & React-Router4 & ES6</h1>
				</header>
			</div>
			)
	}

}

export default Header;

