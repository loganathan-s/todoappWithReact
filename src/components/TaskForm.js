import React from 'react';
import PropTypes from 'prop-types';

class TaskForm extends React.Component{
	

	createTask(event){
		event.preventDefault();
		const task = {task: this.task.value, status: "New"}
		this.props.addTask(task);
		this.taskForm.reset();
	}	

	render(){
		return(
			 <form ref={(input) => this.taskForm = input} className="taskForm"  onSubmit={this.createTask.bind(this)}>
			  <input ref={(input) => this.task = input } type="text" placeholder="Task" />
			  <button type="submit" className="createTask">Add Task</button> 
			 </form>
			)
	}

}


TaskForm.PropTypes = {
  addTask: PropTypes.func.isRequired,
};

export default TaskForm;

