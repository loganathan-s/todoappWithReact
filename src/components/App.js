import React, { Component } from 'react';
import Header from './Header';
import Footer from './Footer';
import TaskForm from './TaskForm';
import Task from './Task';
import logo from '../images/logo.svg';
import {Tasks} from '../simpleTasks';

import '../css/App.css';

class App extends Component {
  constructor(){
    super();
    this.loadSamples = this.loadSamples.bind(this); 
    this.addTask = this.addTask.bind(this); 
    this.removeTask = this.removeTask.bind(this); 
    this.updateTask = this.updateTask.bind(this); 
    this.markComplete = this.markComplete.bind(this); 
    this.setCurrentTaskEditable = this.setCurrentTaskEditable.bind(this); 
    this.state = {
      tasks: {},
      editing: false,
      currentTask: {}
    };
  }

  addTask(task){
    const tasks = {...this.state.tasks};
    const taskKey = Object.keys(tasks).length + 1;
    tasks[`task${taskKey}`] = task;
    this.setState({ tasks })
  }

  removeTask(key){
     const tasks = {...this.state.tasks};
     delete tasks[key];
     this.setState({ tasks }) 
  }

  updateTask(key, updatedTask){
     const tasks = {...this.state.tasks};
     const task = tasks[key]
     task.task = updatedTask
     task.edit = false
     tasks[key] = task
     this.setState({ tasks }) 
  }

  markComplete(key){
     const tasks = {...this.state.tasks};
     let task = tasks[key]
     task.status = (task.status === "Done" ? "InProgress" : "Done")
     tasks[key] = task
     this.setState({ tasks }) 
  }

  setCurrentTaskEditable(key){
    const tasks = {...this.state.tasks};
    const currentTask = tasks[key]
    currentTask["edit"] = !currentTask["edit"];
    tasks[key] = currentTask;
    this.setState({ tasks });
  }

  loadSamples(){
      this.setState({
        tasks: Tasks
      })
  }

  //Callback which runs before rendering the APP component
  componentWillMount(){ 
    this.loadSamples();
  }

    
  render() {
    return (
      <div className="App">
         <Header />
          <div className="content">
            <TaskForm addTask={this.addTask} /> 
            <ul className="allTasks">
              { Object.
                keys(this.state.tasks).map(
                  key => <Task key={key} index={key} task={this.state.tasks[key]} removefromTask={this.removeTask} markasDone={this.markComplete}  setCurrentTaskEditable={this.setCurrentTaskEditable} updateTask={this.updateTask}/>)
            }
            </ul>
          </div>
        <Footer/>
      </div>
    );
  }
}

export default App;
