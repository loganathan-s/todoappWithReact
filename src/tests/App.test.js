import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../components/App';
import Header from '../components/Header';
import TaskForm from '../components/TaskForm';
import Task from '../components/Task';
import Footer from '../components/Footer';

describe(App, () => {
	// Added to allow lifecycles, bydefault Enzyme wont execute lifecylce methods
	const options = {
  		lifecycleExperimental: true,
  		disableLifecycleMethods: false 
	};

  const component = shallow(<App />, options);

  it('renders and matches our snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains an Header subcomponent', () => {
    expect(component.find(Header)).toHaveLength(1);
  });

  it('contains an TaskForm subcomponent', () => {
    expect(component.find(TaskForm)).toHaveLength(1);
  });

  it('contains an Task subcomponent', () => {
    expect(component.find(Task)).toHaveLength(9);
  });

  it('contains an Footer subcomponent', () => {
    expect(component.find(Footer)).toHaveLength(1);
  });

  it('contains the same number of Task components as tasks', () => {
  	const tasks = component.find(Task).length;
  	const s_tasks   = Object.keys(component.state('tasks')).length;
  	expect(tasks).toEqual(s_tasks);
  });

});
