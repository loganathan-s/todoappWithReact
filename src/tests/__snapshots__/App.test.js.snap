// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class App extends _react.Component {
  constructor() {
    super();
    this.loadSamples = this.loadSamples.bind(this);
    this.addTask = this.addTask.bind(this);
    this.removeTask = this.removeTask.bind(this);
    this.updateTask = this.updateTask.bind(this);
    this.markComplete = this.markComplete.bind(this);
    this.setCurrentTaskEditable = this.setCurrentTaskEditable.bind(this);
    this.state = {
      tasks: {},
      editing: false,
      currentTask: {} };

  }

  addTask(task) {
    const tasks = Object.assign({}, this.state.tasks);
    const taskKey = Object.keys(tasks).length + 1;
    tasks[\`task\${taskKey}\`] = task;
    this.setState({ tasks });
  }

  removeTask(key) {
    const tasks = Object.assign({}, this.state.tasks);
    delete tasks[key];
    this.setState({ tasks });
  }

  updateTask(key, updatedTask) {
    const tasks = Object.assign({}, this.state.tasks);
    const task = tasks[key];
    task.task = updatedTask;
    task.edit = false;
    tasks[key] = task;
    this.setState({ tasks });
  }

  markComplete(key) {
    const tasks = Object.assign({}, this.state.tasks);
    let task = tasks[key];
    task.status = task.status === "Done" ? "InProgress" : "Done";
    tasks[key] = task;
    this.setState({ tasks });
  }

  setCurrentTaskEditable(key) {
    const tasks = Object.assign({}, this.state.tasks);
    const currentTask = tasks[key];
    currentTask["edit"] = !currentTask["edit"];
    tasks[key] = currentTask;
    this.setState({ tasks });
  }

  loadSamples() {
    this.setState({
      tasks: _simpleTasks.Tasks });

  }

  //Callback which runs before rendering the APP component
  componentWillMount() {
    this.loadSamples();
  }


  render() {
    return (
      _react2.default.createElement('div', { className: 'App', __source: { fileName: _jsxFileName, lineNumber: 79 }, __self: this },
        _react2.default.createElement(_Header2.default, { __source: { fileName: _jsxFileName, lineNumber: 80 }, __self: this }),
        _react2.default.createElement('div', { className: 'content', __source: { fileName: _jsxFileName, lineNumber: 81 }, __self: this },
          _react2.default.createElement(_TaskForm2.default, { addTask: this.addTask, tasks: this.state.tasks, __source: { fileName: _jsxFileName, lineNumber: 82 }, __self: this }),
          _react2.default.createElement('ul', { className: 'allTasks', __source: { fileName: _jsxFileName, lineNumber: 83 }, __self: this },
            Object.
            keys(this.state.tasks).map(
            key => _react2.default.createElement(_Task2.default, { key: key, index: key, task: this.state.tasks[key], removefromTask: this.removeTask, markasDone: this.markComplete, setCurrentTaskEditable: this.setCurrentTaskEditable, updateTask: this.updateTask, __source: { fileName: _jsxFileName, lineNumber: 86 }, __self: this })))),



        _react2.default.createElement(_Footer2.default, { __source: { fileName: _jsxFileName, lineNumber: 90 }, __self: this })));


  }} renders and matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <header
      className="top"
    >
      <h1>
         This is a simple TODO application, developed using React 15.6 & React-Router4 & ES6
      </h1>
    </header>
  </div>
  <div
    className="content"
  >
    <form
      className="taskForm"
      onSubmit={[Function]}
    >
      <input
        placeholder="Task"
        type="text"
      />
      <button
        type="submit"
      >
        Add Task
      </button>
    </form>
    <ul
      className="allTasks"
    >
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          Everyones favorite white task. We will cut it to the size you need and ship it.
           - 
          in_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          These tender, mouth-watering beauties are a fantastic hit at any dinner party.
           - 
          in_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          Big, sweet and tender. True dry-pack scallops from the icey waters of Alaska. About 8-10 per pound
           - 
          unin_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          Lean flesh with a mild, sweet flavor profile, moderately firm texture and large, moist flakes. 
           - 
          in_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          Crack these open and enjoy them plain or with one of our cocktail sauces
           - 
          in_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          This flaky, oily salmon is truly the king of the sea. Bake it, grill it, broil it...as good as it gets!
           - 
          in_progress
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          A soft plump oyster with a sweet salty flavor and a clean finish.
           - 
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          The best mussels from the Pacific Northwest with a full-flavored and complex taste.
           - 
          done
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
      <li
        className="task"
      >
        <h3
          className="taskName"
        >
          With 21-25 two bite prawns in each pound, these sweet morsels are perfect for shish-kabobs.
           - 
          new
        </h3>
        <div>
          <button
            className="markcomplete"
            onClick={[Function]}
          >
            Mark Complete
          </button>
          <button
            className="editTask"
            onClick={[Function]}
          >
            Edit Task
          </button>
          <button
            className="removeTask"
            onClick={[Function]}
          >
            RemoveTask
          </button>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="footer"
  >
    <footer
      className="footer"
    >
      <p>
         Footer
      </p>
    </footer>
  </div>
</div>
`;
